git/odb/object_type.go

```diff
@@ -0,0 +1,45 @@
+package odb
+
+import "strings"
+
+// ObjectType is a constant enumeration type for identifying the kind of object
+// type an implementing instance of the Object interface is.
+type ObjectType uint8
+
+const (
+	UnknownObjectType ObjectType = iota
+	BlobObjectType
+	TreeObjectType
+	CommitObjectType
+)
+
+// ObjectTypeFromString converts from a given string to an ObjectType
+// enumeration instance.
+func ObjectTypeFromString(s string) ObjectType {
+	switch strings.ToLower(s) {
+	case "blob":
+		return BlobObjectType
+	case "tree":
+		return TreeObjectType
+	case "commit":
+		return CommitObjectType
+	default:
+		return UnknownObjectType
+	}
```

Comment:
You may also encounter tags in the wild. It probably makes sense to ignore them for now, though, and get the basic algorithm working.